### PR TITLE
Improve type of Statement#parent

### PR DESCRIPTION
**Pull request by andy-ms**
Unfortunately, `MissingDeclaration` multiply inherits from `DeclarationStatement` and from three other types that aren't statements. So had to create a `StatementOnly` type for things that only occur in statement position.

### CODE REVIEW

File: src/compiler/types.ts [hunk 1]
```diff
@@ -1832,17 +1837,22 @@ namespace ts {
         _statementBrand: any;
     }
 
+    /** Type for statements that appear only in statement position. */
+    export interface StatementOnly extends Statement {
+        parent?: Block | CaseClause | DefaultClause | SourceFile | ModuleBlock | LabeledStatement;
```
Comment (ajafff) [inline src/compiler/types.ts]: + IfStatement | WithStatement | IterationStatement

Comment (ajafff) [reply src/compiler/types.ts]: Could also use BlockLike

**Comment by andy-ms** [comment]
No longer fixes #22027 as I have done that in #22131 instead.

File: src/compiler/types.ts [hunk 2]
```diff
@@ -1832,17 +1837,22 @@ namespace ts {
         _statementBrand: any;
     }
 
+    /** Type for statements that appear only in statement position. */
+    export interface StatementOnly extends Statement {
```
Comment (mhegazy) [inline src/compiler/types.ts]: Can we split off MissingDeclaration instead of having to create this type?

Comment (andy-ms) [reply src/compiler/types.ts]: Turns out MissingDeclaration has some supertypes it doesn't need. See #23485.

**Pull request closed without merging**